packages/block-editor/src/components/global-styles/typography-panel.js

```diff
@@ -176,17 +176,42 @@ export default function TypographyPanel( {
 	// Font Size
 	const hasFontSizeEnabled = useHasFontSizeControl( settings );
 	const disableCustomFontSizes = ! settings?.typography?.customFontSize;
-	const mergedFontSizes = uniqByProperty(
-		mergeOrigins( {
-			default:
-				settings?.typography?.defaultFontSizes !== false
-					? settings?.typography?.fontSizes?.default
-					: undefined,
-			theme: settings?.typography?.fontSizes?.theme,
-			custom: settings?.typography?.fontSizes?.custom,
-		} ),
+
+	/**
+	 * TODO: Doing some hacky things here so the global styles CSS matches what
+	 * is shown in the dropdown.
+	 *
+	 * Global styles PRESETS_METADATA includes a `prevent_override` option
+	 * that's supposed to handle this. But it isn't working as expected.
```

Comment:
I'm now not convinced that's what `prevent_override` is meant to do. I think the purpose of that is to allow the different settings to be split out onto different levels within the array created by theme.json (i.e `default`, `theme` and `custom`). This prevents the theme from overriding the defaults.

The question of what happens to duplicates not dealt with by Global Styles I think, but the behaviour seems consistent - whatever the preset is, is kept and cannot be overridden by the theme. I think we should keep this behaviour, which means I think we can merge this PR, we just need to update this comment.

Reply:
My comment was just poorly worded. I opened a separate issue for a follow-up (#57733) and updated the comment to be a bit more clear that the hack is to update the UI to match the buggy stylesheet.